sources/models/src/modeled_types/shared.rs

```diff
@@ -250,3 +252,99 @@ mod test_url {
         }
     }
 }
+
+// =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=
+
+/// FriendlyVersion represents a valid version string that can be prefixed with or without 'v.
+/// It can also be set to 'latest' to represent the latest version. It stores the original string
+/// and makes it accessible through standard traits.
+#[derive(Debug, Clone, Eq, PartialEq, Hash)]
+pub struct FriendlyVersion {
+    inner: String,
+}
+
+impl TryFrom<&str> for FriendlyVersion {
+    type Error = error::Error;
+
+    fn try_from(input: &str) -> Result<Self, Self::Error> {
+        if input == "latest" {
+            return Ok(FriendlyVersion {
+                inner: input.to_string(),
+            });
+        }
+        // If the string begins with a 'v', skip it before checking if it is valid semver.
+        let version = if input.starts_with('v') {
+            &input[1..]
+        } else {
+            input
+        };
+
+        if version.parse::<semver::Version>().is_ok() {
+            return Ok(FriendlyVersion {
+                inner: input.to_string(),
+            });
+        }
+        error::InvalidVersion { input }.fail()
+    }
+}
+
+impl TryInto<semver::Version> for FriendlyVersion {
+    type Error = semver::SemVerError;
+
+    fn try_into(self) -> Result<Version, Self::Error> {
+        // If the string begins with a 'v', skip it before conversion
+        let version = if self.inner.starts_with('v') {
+            &self.inner[1..]
+        } else {
+            &self.inner
+        };
+        Version::from_str(version)
+    }
+}
+
+string_impls_for!(FriendlyVersion, "FriendlyVersion");
+
+#[cfg(test)]
+mod test_version {
+    use super::FriendlyVersion;
+    use semver::{SemVerError, Version};
+    use std::convert::TryFrom;
+    use std::convert::TryInto;
+
+    #[test]
+    fn good_version_strings() {
+        for ok in &[
+            "1.0.0",
+            "v1.0.0",
+            "1.0.1-alpha",
+            "v1.0.1-alpha",
+            "1.0.2-alpha+1.0",
+            "v1.0.2-alpha+1.0",
+            "1.0.3-beta.1.01",
+            "v1.0.3-beta.1.01",
+            "1.1.0-rc.1.1",
+            "v1.1.0-rc.1.1",
+            "latest",
+        ] {
+            FriendlyVersion::try_from(*ok).unwrap();
+            // Test conversion to semver::Version
+            if *ok != "latest" {
+                let _: Version = FriendlyVersion::try_into(FriendlyVersion {
```

Comment:
Same in the test below this.  Also, you might be able to do a similar change to keep things consistent, `Version::try_from(FriendlyVersion { bla })` rather than bringing in TryInto.  (Not that TryInto is bad, but TryFrom can make it clearer what your goal type is, and we're already using it for more other pieces.)